### PR TITLE
Adds path parameter to NodeMatch

### PR DESCRIPTION
**Pull request by reinvanimschoot**
#### Is this adding or improving a _feature_ or fixing a _bug_?

Feature

#### What's the new behavior?

The NodeMatch function provided in several functions on the  `Editor` and `Transform` now takes a second parameter which is the path of the node provided as the first parameter.

#### How does this change work?

This is a non breaking change.

In the editor.ts, the `Editor.levels` and `Editor.nodes` functions are the only two that explicitly call `match`, so these have been updated to account for the `nodePath` as the second parameter.

In transforms/node.ts, all match functionality is deferred to one of the two functions above but they had a different type, not `NodeMatch<T>` but `(node: node) => boolean`. This has now been updated as well.

#### Have you checked that...?

- [X] The new code matches the existing patterns and styles.
- [X] The tests pass with `yarn test`.
- [X] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [X] The relevant examples still work. (Run examples with `yarn start`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: #3967
Reviewers: @BrentFarese

### CODE REVIEW

File: packages/slate/src/interfaces/editor.ts
Comment (BrentFarese) [inline packages/slate/src/interfaces/editor.ts:1578]: Will this type break existing implementations that have a `NodeMatch` w/o a `path` parameter?  Does everyone now need to update their `match` functions?  I think no but figured I'd double-check...

**Review by BrentFarese — APPROVED** [review]
I think this looks good.  @thesunny or @CameronAckermanSEL care to comment?

**Comment by BrentFarese** [comment]
Note: This PR overlaps with #3968 and after merging we can close #3968.

File: packages/slate/src/interfaces/editor.ts
Comment (reinvanimschoot) [reply packages/slate/src/interfaces/editor.ts:1578]: It shouldn't, adding a second possible parameter like this does not necessarily mean you have to pass a second argument every time. Like when you call a `forEach`, you can call it with the element and index or just the element.

But if you really want me to double check, I can. Is there a simple way to make this version of the code run in my own project?

Comment (BrentFarese) [reply packages/slate/src/interfaces/editor.ts:1578]: No it should be fine there's ton's of `match` functions in Slate core with just `node` so TS would've thrown an error in Slate core is there was an issue..

**Comment by reinvanimschoot** [comment]
@BrentFarese what are the following steps for this to become merged if I may ask?

**Comment by BrentFarese** [comment]
> @BrentFarese what are the following steps for this to become merged if I may ask?

I would like at least some time for @thesunny or @CameronAckermanSEL or another maintainer(s) to review, then I can merge this in.  It's not a controversial change and seems good, just want an additional set of eyes.

**Comment by reinvanimschoot** [comment]
I understand that a second pair of eyes is desired before merging but I'm a bit woried that this PR will suffer the same fate as the countless other PR's that are still open on this project. I fixed this code specifically because we have an urgent need for it in our own project. So I really hope another maintainer can take a look soon.

**Comment by ianstormtaylor** [comment]
@BrentFarese it looks good to me. (But I haven't tested it locally.)

**Pull request merged**
